snap/README.md

```diff
@@ -0,0 +1,167 @@
+
+```
+$ sudo snap connect edgex-device-snmp:device-config your-snap:device-config
+```
+
+This needs to be done before the device service is started for the first time. Once you have set the configuration the device service can be started and it will then be configurated using the settings you provided:
+
+```
+$ sudo snap start edgex-device-snmp.device-snmp
+```
+
+**Note** - content interfaces from snaps installed from the Snap Store that have the same publisher connect automatically. For more information on snap content interfaces please refer to the snapcraft.io [Content Interface](https://snapcraft.io/docs/content-interface) documentation.
+
+### Autostart
+
+By default, the edgex-device-snmp disables its service on install, as the expectation is that the default profile configuration files will be customized, and thus this behavior allows the profile `configuration.toml` files in $SNAP_DATA to be modified before the service is first started.
+
+This behavior can be overridden by setting the `autostart` configuration setting to "true". This is useful when configuration and/or device profiles are being provided via configuration or gadget snap content interface.
+
+**Note** - this option is typically set from a gadget snap.
+
+### Rich Configuration
+
+While it's possible on Ubuntu Core to provide additional profiles via gadget snap content interface, quite often only minor changes to existing profiles are required.
+
+These changes can be accomplished via support for EdgeX environment variable configuration overrides via the snap's configure hook. If the service has already been started, setting one of these overrides currently requires the service to be restarted via the command-line or snapd's REST API. If the overrides are provided via the snap configuration defaults capability of a gadget snap, the overrides will be picked up when the services are first started.
+
+The following syntax is used to specify service-specific configuration overrides:
+
+```
+env.<stanza>.<config option>
+```
+
+For instance, to setup an override of the service's Port use: 
+
+```
+$ sudo snap set edgex-device-snmp env.service.port=2112 
+```
+
+And restart the service: 
+
+```
+$ sudo snap restart edgex-device-snmp.device-snmp
+```
+
+**Note** - at this time changes to configuration values in the [Writable] section are not supported. For details on the mapping of configuration options to Config options, please refer to "Service Environment Configuration Overrides".
+
+## Service Environment Configuration Overrides
+
+**Note** - all of the configuration options below must be specified with the prefix: `env.`
+
+```
+[Service]
+service.boot-timeout            // Service.BootTimeout
+service.health-check-interval   // Service.HealthCheckInterval
+service.host                    // Service.Host
+service.server-bind-addr        // Service.ServerBindAddr
+service.port                    // Service.Port
+service.protocol                // Service.Protocol
```

Comment:
`Service.Protocol` is probably a deprecated config argument. Not supported by the snap.

Reply:
Removed.